egui/src/widgets/plot/mod.rs

```diff
@@ -1094,3 +1212,36 @@ impl PreparedPlot {
         }
     }
 }
+
+/// Returns next bigger power in given base
+/// e.g.
+/// ```ignore
+/// next_power(0.01, 10) == 0.01
+/// next_power(0.02, 10) == 0.1
+/// next_power(0.2,  10) == 1
+/// ```
+fn next_power(value: f64, base: f64) -> f64 {
+    assert_ne!(value, 0.0); // can be negative (typical for Y axis)
+    base.powi(value.abs().log(base).ceil() as i32)
+}
+
+/// Fill in all values between [min, max] which are a multiple of `step_size`
+fn generate_marks(step_sizes: [f64; 3], bounds: (f64, f64)) -> Vec<GridMark> {
+    let mut steps = vec![];
+    fill_marks_between(&mut steps, step_sizes[0], bounds);
+    fill_marks_between(&mut steps, step_sizes[1], bounds);
+    fill_marks_between(&mut steps, step_sizes[2], bounds);
+    steps
+}
+
+/// Fill in all values between [min, max] which are a multiple of `step_size`
+fn fill_marks_between(out: &mut Vec<GridMark>, step_size: f64, (min, max): (f64, f64)) {
+    assert!(max > min);
+    let first = (min / step_size).ceil() as i64;
+    let last = (max / step_size).ceil() as i64;
+
+    for i in first..last {
+        let value = (i as f64) * step_size;
+        out.push(GridMark { value, step_size });
+    }
```

Comment:
using `out.extend` should be faster than repeatedly calling `.push`

Reply:
Done (needed conversion into an iterator).